psydac/feec/multipatch/examples_nc/hcurl_eigen_pbms_nc.py

```diff
@@ -0,0 +1,379 @@
+"""
+    Solve the eigenvalue problem for the curl-curl operator in 2D with non-matching FEEC discretization
+"""
+import os
+from mpi4py import MPI
+
+import numpy as np
+import matplotlib.pyplot as plt
+from collections import OrderedDict
+from sympde.topology import Derham
+
+from psydac.feec.multipatch.api import discretize
+from psydac.api.settings import PSYDAC_BACKENDS
+from psydac.feec.multipatch.fem_linear_operators import IdLinearOperator
+from psydac.feec.multipatch.operators import HodgeOperator
+from psydac.feec.multipatch.multipatch_domain_utilities import build_multipatch_domain
+from psydac.feec.multipatch.plotting_utilities import plot_field
+from psydac.feec.multipatch.utilities import time_count, get_run_dir, get_plot_dir, get_mat_dir, get_sol_dir, diag_fn
+from psydac.feec.multipatch.utils_conga_2d import write_diags_to_file
+
+from sympde.topology import Square
+from sympde.topology import IdentityMapping, PolarMapping
+from psydac.fem.vector import ProductFemSpace
+
+from scipy.sparse.linalg import spilu, lgmres
+from scipy.sparse.linalg import LinearOperator, eigsh, minres
+from scipy.sparse import csr_matrix
+from scipy.linalg import norm
+
+from psydac.linalg.utilities import array_to_psydac
+from psydac.fem.basic import FemField
+
+from psydac.feec.multipatch.non_matching_multipatch_domain_utilities import create_square_domain
+from psydac.feec.multipatch.non_matching_operators import construct_hcurl_conforming_projection
+
+from psydac.api.postprocessing import OutputManager, PostProcessManager
+
+
+def hcurl_solve_eigen_pbm_nc(ncells=np.array([[8, 4], [4, 4]]), degree=(3, 3), domain=([0, np.pi], [0, np.pi]), domain_name='refined_square', backend_language='pyccel-gcc', mu=1, nu=0, gamma_h=0,
+                             generalized_pbm=False, sigma=5, ref_sigmas=None, nb_eigs_solve=8, nb_eigs_plot=5, skip_eigs_threshold=1e-7,
+                             plot_dir=None, hide_plots=True, m_load_dir=None,):
+    """
+    Solve the eigenvalue problem for the curl-curl operator in 2D with DG discretization
```

Comment:
same question as above. Also, what is the difference with the function `hcurl_solve_eigen_pbm_dg` in `psydac/feec/multipatch/examples_nc/hcurl_eigen_pbms_dg.py`? Is that the same scheme with a non-conforming (non matching) grid?

Reply:
This is a typo in the doc string. This function is a non-matching broken FEEC scheme.